### ISSUE TITLE
Crash when using rake examples:convert with highlight.js

### ISSUE THREAD

**Issue by obilodeau** [asciidoctor/asciidoctor-reveal.js#338]
I was preparing some slide deck to announce the release and I experienced an issue with the rake task `examples:convert`.

I reduced it to a very simple example (derived from `examples/listing.adoc`):
```asciidoc
= Presentation
Me
:source-highlighter: highlight.js
:icons: font

== Listing

----
fn main() {
    println!("Hello World!");
}
----
```

Top of the stack trace:
```
Converting file examples/listing.adoc... rake aborted!
NoMethodError: undefined method `merge' for nil:NilClass
/home/olivier/src/asciidoc/asciidoctor-reveal.js/lib/asciidoctor-revealjs/highlightjs.rb:30:in `format'
/home/olivier/src/asciidoc/asciidoctor-reveal.js/templates/listing.html.slim:19:in `__tilt_3060'
/home/olivier/src/asciidoc/asciidoctor-reveal.js/.bundle/gems/ruby/2.7.0/gems/tilt-2.0.10/lib/tilt/template.rb:170:in `call'
/home/olivier/src/asciidoc/asciidoctor-reveal.js/.bundle/gems/ruby/2.7.0/gems/tilt-2.0.10/lib/tilt/template.rb:170:in `evaluate'
/home/olivier/src/asciidoc/asciidoctor-reveal.js/.bundle/gems/ruby/2.7.0/gems/tilt-2.0.10/lib/tilt/template.rb:109:in `render'
/home/olivier/src/asciidoc/asciidoctor-reveal.js/.bundle/gems/ruby/2.7.0/gems/asciidoctor-2.0.10/lib/asciidoctor/converter/template.rb:109:in `convert'
/home/olivier/src/asciidoc/asciidoctor-reveal.js/.bundle/gems/ruby/2.7.0/gems/asciidoctor-2.0.10/lib/asciidoctor/converter/composite.rb:28:in `convert'
/home/olivier/src/asciidoc/asciidoctor-reveal.js/.bundle/gems/ruby/2.7.0/gems/asciidoctor-2.0.10/lib/asciidoctor/abstract_block.rb:75:in `convert'
/home/olivier/src/asciidoc/asciidoctor-reveal.js/.bundle/gems/ruby/2.7.0/gems/asciidoctor-2.0.10/lib/asciidoctor/abstract_block.rb:84:in `block in content'
/home/olivier/src/asciidoc/asciidoctor-reveal.js/.bundle/gems/ruby/2.7.0/gems/asciidoctor-2.0.10/lib/asciidoctor/abstract_block.rb:84:in `map'
/home/olivier/src/asciidoc/asciidoctor-reveal.js/.bundle/gems/ruby/2.7.0/gems/asciidoctor-2.0.10/lib/asciidoctor/abstract_block.rb:84:in `content'
```

This doesn't affect the `asciidoctor-revealjs` CLI unless `-T templates/` is used. Because of this, I'm not sure if this is an upstream bug or not? @Mogztter, @mojavelinux. 

But why would something be `nil` when using the rake task and `{}` when using the compiled templates?

**Comment by Mogztter** [comment]
@obilodeau The code is slightly different in `listing.html.slim `hence the error.
The code should be:

```slim
- if @style == 'source'
  - syntax_hl = document.syntax_highlighter
```

Otherwise `syntax_hl` is defined but `opts` is not.
Reference implementation: https://github.com/asciidoctor/asciidoctor/blob/master/lib/asciidoctor/converter/html5.rb#L655

**Comment by obilodeau** [comment]
My current fix was uglier (setting `opts = {}` in the `else` block). Thanks!

**Comment by Mogztter** [comment]
> My current fix was uglier (setting opts = {} in the else block). Thanks!

It's my bad 😅 
I think we should add a test with a listing block:

```
[listing]
This is an example of a paragraph styled with `listing`.
Notice that the monospace markup is preserved in the output.
```

and check that the syntax highlighter is not applied to this block.

**Comment by obilodeau** [comment]
For the test, the listing block must be present **and** highlight.js activated. We had each individually but not both. I added one here: 58f7da5b. I'm going to refine the text with yours.

**Comment by obilodeau** [comment]
Fixed by #339
